lib/matplotlib/tests/test_category.py

```diff
@@ -269,3 +270,23 @@ def test_mixed_type_update_exception(self, ax, plotter, xdata):
         with pytest.raises(TypeError):
             plotter(ax, [0, 3], [1, 3])
             plotter(ax, xdata, [1, 2])
+
+
+@pytest.mark.style('default')
+@check_figures_equal(extensions=["png"])
+def test_overliding_units(fig_test, fig_ref):
+    from datetime import datetime
+
+    t0 = datetime(2018, 3, 1)
+    t1 = datetime(2018, 3, 2)
+    t2 = datetime(2018, 3, 3)
+    t3 = datetime(2018, 3, 4)
+
```

Comment:
what's the point of adding in the datetime here if that's not what's being tested?

Reply:
It is a converter class that does not care about the value of `Axis.units`, but I think it is worth putting units on both Axis.

Reply:
Would it be better to test using some data that does care about/use units? I'm wondering if datetime always has the units set to `None` all the time anyway.

Reply:
The string catarogicals do use `Axis.units` to store critical state (which is what broke in #11095).

Reply:
Oh woops, I missed that categoricals were being plotted too

Reply:
@tacaswell so that's what concerns me, that you're putting two potential points of failure in this test - when it fails will it be easy to see that it was the categoricals that broke this?

Reply:
All of our tests as pretty much closer to integration tests than unit tests.   I am optimistic that if (when) this break who ever is debugging it will be able to sort out the root cause with out too much trouble.